src/Mod/PartDesign/Gui/ViewProviderHole.cpp

```diff
@@ -49,10 +49,14 @@ ViewProviderHole::ViewProviderHole()
 
 ViewProviderHole::~ViewProviderHole() = default;
 
-std::vector<App::DocumentObject*> ViewProviderHole::claimChildren()const
+std::vector<App::DocumentObject*> ViewProviderHole::claimChildren() const
 {
     std::vector<App::DocumentObject*> temp;
-    temp.push_back(getObject<PartDesign::Hole>()->Profile.getValue());
+
+    if (App::DocumentObject* profile = getObject<PartDesign::Hole>()->Profile.getValue();
+        !profile->isDerivedFrom<PartDesign::Feature>()) {

```

Comment:
No need to check if `profile` is `nullptr`?

Reply:
Right, should be `profile &&`